plugins/inputs/win_perf_counters/pdh.go

```diff
@@ -417,3 +432,16 @@ func UTF16PtrToString(s *uint16) string {
 	}
 	return syscall.UTF16ToString((*[1 << 29]uint16)(unsafe.Pointer(s))[0:])
 }
+
+func UTF16ToStringArray(buf []uint16) []string {
+	var strings []string
+	nextLineStart := 0
+	stringLine := UTF16PtrToString(&buf[0])
+	for stringLine != "" {
+		strings = append(strings, stringLine)
+		nextLineStart += len(stringLine) + 1
```

Comment:
This looks very fishy to me, the length of `stringLine` tells us nothing about the offset into buf, because buf contains utf16 while stringLine contains utf8.